airbyte-integrations/base-java/src/main/java/io/airbyte/integrations/base/DestinationConsumer.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.base;
+
+import io.airbyte.commons.functional.CheckedConsumer;
+
+/**
+ * Lifecyle: * Consumer object is instantiated. * It received messages via
+ * {@link DestinationConsumer#accept(T)} * Upon receiving the last message (assuming no failures)
+ * {@link DestinationConsumer#complete()} * Always (on success or failure) finalize by calling
```

Comment:
Now that we always have to call complete. I think we coud just use close and remove complete completely.

Reply:
Since `close` is called wether it is successful or not and same for `complete`, `complete` could be called in `close` thus isn't necessary. It might actually be a source of bug since it is not going to be call in the try-resource.

Reply:
the contract hasn't changed; this is the javaspotless reformatting making this impossible to read. it is reformatting the comment new lines. here's a clearer version

```
 * Lifecyle: 1. Consumer object is instantiated. 2. It received messages via
 * {@link DestinationConsumer#accept(T)} 3. Upon receiving the last message (assuming no failures)
 * {@link DestinationConsumer#complete()} 4. Always (on success or failure) finalize by calling
 * {@link DestinationConsumer#close()}
```

Reply:
discussed offline. removing complete and will force integrations to track internal state if needed. will add helpers for this as we encounter it.